src/semidiscretization/semidiscretization_coupled.jl

```diff
@@ -452,21 +477,42 @@ function copy_to_coupled_boundary!(boundary_condition, u_ode, semi)
     return nothing
 end
 
+function copy_to_coupled_boundary!(u_ode, semi, i, n_boundaries, boundary_condition,
+                                   boundary_conditions...)
+    copy_to_coupled_boundary!(boundary_condition, u_ode, semi)
```

Comment:
`i` is not used here? It looks like this is just the same call in every loop. Also, you could use `foreach_enumerate` here as well.

Reply:
`i` is used as break condition:
```
function copy_to_coupled_boundary!(u_ode, semi, i, n_boundaries, boundary_condition,
                                   boundary_conditions...)
    copy_to_coupled_boundary!(boundary_condition, u_ode, semi)
    if i < n_boundaries
        copy_to_coupled_boundary!(u_ode, semi, i + 1, n_boundaries,
                                  boundary_conditions...)
    end
end
```

Reply:
Ah right, sorry.